components/mediation-admin/org.wso2.carbon.localentry/src/main/java/org/wso2/carbon/localentry/service/LocalEntryAdmin.java

```diff
@@ -256,10 +266,15 @@ public boolean saveEntry(String ele) throws LocalEntryAdminException {
                     getSynapseConfiguration().removeEntry(key);
                     getSynapseConfiguration().addEntry(key, entry);
                     entry.setFileName(oldEntry.getFileName());
-                    MediationPersistenceManager pm
-                            = ServiceBusUtils.getMediationPersistenceManager(getAxisConfig());
-                    pm.saveItem(key, ServiceBusConstants.ITEM_TYPE_ENTRY);
-
+                    CAppArtifactDataService cAppArtifactDataService = ConfigHolder.getInstance().
+                            getcAppArtifactDataService();
+                    if (cAppArtifactDataService.isArtifactDeployedFromCApp(getTenantId(), ServiceBusConstants.LOCAL_ENTRY_TYPE + File.separator + key)) {
+                        cAppArtifactDataService.setEdited(getTenantId(), ServiceBusConstants.LOCAL_ENTRY_TYPE + File.separator + key);
```

Comment:
ServiceBusConstants.LOCAL_ENTRY_TYPE + File.separator + key - make this a variable and use in multiple places